app/controllers/ad_steps_controller.rb

```diff
@@ -0,0 +1,58 @@
+class AdStepsController < ApplicationController
+  include Wicked::Wizard
+  steps :details, :picture, :contact
+  before_action :find_ad, only: [:show, :update]
+
+  def show
+    render_wizard
+  end  
+  
+  def update
+    case step
+    when :picture
+      @ad.update(details_params)
+    when :contact
+      @ad.update(image_params)
+    end
+    if params[:id] == 'wicked_finish'
+      @ad.update(contact_params)
+      render "wicked_finish"
+    else
+      render_wizard
+    end
+  end
+
+  def wicked_finish
+  end
+
+
+  private
+  
+  def find_ad
+    @ad = Ad.new if @ad.blank?
```

Comment:
Can we test this after removing this line?

Reply:
Yes!